tensorflow/core/kernels/data/captured_function.h

```diff
@@ -221,7 +221,8 @@ class InstantiatedCapturedFunction {
   // possible.
   Status RunWithBorrowedArgs(IteratorContext* ctx,
                              const std::vector<Tensor>& args,
-                             std::vector<Tensor>* rets) const;
+                             std::vector<Tensor>* rets,
```

Comment:
Please add an overload for `Run` and `RunWithBorrowedArgs` which does not take `Node` (i.e. matching the signature of the method prior to your change), which calls into the newly introduced API, setting the argument to `nullptr`. That way, you PR will be able to merge cleanly with internal version of TensorFlow without having to updated any internal callers of `Run` and `RunWithBorrowedArgs` synchronously with submitting your OR.

Reply:
I think Default args are the most clean way to express this here, since this is a non-virtual function and overloading adds the same logical boilerplate.

Reply:
Ah, just saw the other comment. Will fix.

Reply:
Default arguments are strongly discouraged by Google C++ style guide even for non-virtual functions.

Reply:
https://google.github.io/styleguide/cppguide.html#Default_Arguments

Reply:
For tf.data code we heed this advice and consistency within tf.data code base outweighs the lack of boilerplate default arguments provide.